index.d.ts

```diff
@@ -63,14 +64,10 @@ import elementReady = require('element-ready');
 })();
 ```
 */
-declare function elementReady<ElementName extends keyof HTMLElementTagNameMap>(
-	selector: ElementName,
+declare function elementReady<Selector extends string, ElementName extends Element = ParseSelector<Selector>>(
+	selector: Selector,
 	options?: elementReady.Options
-): elementReady.StoppablePromise<HTMLElementTagNameMap[ElementName] | undefined>;
-declare function elementReady<ElementName extends keyof SVGElementTagNameMap>(
-	selector: ElementName,
-	options?: elementReady.Options
-): elementReady.StoppablePromise<SVGElementTagNameMap[ElementName] | undefined>;
+): elementReady.StoppablePromise<ElementName | undefined>;
 declare function elementReady<ElementName extends Element = Element>(
```

Comment:
Is this now redundant?

Reply:
I left this as a fallback [like in `select-dom`](https://github.com/fregante/select-dom/blob/93c48807aa37333a1b89c962f19863f151a3aade/index.ts#L26) since, as I understood it, `ParseSelector` [can return `never`](https://github.com/g-plane/typed-query-selector/blob/5147a5d562adf7bca920b2ba711d21f5477569b8/parser.d.ts#L116).